### PR TITLE
v2: Introduce crate, define format

### PR DESCRIPTION
**Pull request by rvolosatovs**
Groundwork for #17:
- introduce a new re-exported `v2` sub-crate
- define shared `Result` and `Error` types
- define shared block format items
- document block format

Docs currently available at: https://rvolosatovs.github.io/universe/rust/sallyport/index.html

### CODE REVIEW

File: src/v2/src/item/mod.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod syscall;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[repr(usize)]
+pub enum Kind {
+    End = 0x00,
```
Comment (rvolosatovs) [inline src/v2/src/item/mod.rs:8]: We could silence these, but this is going to be used in follow-up PRs, so not sure we should

File: src/v2/src/item/syscall.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[repr(C, align(8))]
+pub struct Header {
+    pub item: super::Header,
```
Comment (haraldh) [inline src/v2/src/item/syscall.rs:6]: why include the super Header in here??

**Review by haraldh — CHANGES_REQUESTED** [review]

File: src/v2/src/item/syscall.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    pub num: usize,
+}
```
Comment (haraldh) [inline src/v2/src/item/syscall.rs:8]: I would have expected:
```suggestion
pub struct Header {    
    pub num: usize,
    pub args: [usize; 6],
}
```

File: src/v2/src/item/syscall.rs (hunk 2 above)
Comment (rvolosatovs) [reply src/v2/src/item/syscall.rs:6]: Each item's contents are prefixed by item's size and kind.
Each syscall is an item.

Practically speaking that means that you can just read or write one `syscall::Header` in one `ptr::read` or `ptr::write`

Comment (haraldh) [reply src/v2/src/item/syscall.rs:6]: @rvolosatovs You don't do that before you haven't parsed the `kind`.

File: src/v2/src/item/syscall.rs (hunk 3 above)
Comment (rvolosatovs) [reply src/v2/src/item/syscall.rs:8]: This depends on what you define as "header", I'd say that `args` and `ret` are "payload" and, in fact, the way that payload is handled depends on the `num`, which is contained in the header.
For example, the host only needs to read 3 arguments from argv on a read and it should not read return values at all.

File: src/v2/src/item/syscall.rs (hunk 2 above)
Comment (rvolosatovs) [reply src/v2/src/item/syscall.rs:6]: On read yes, indeed, but on `write` it is clearer and simpler to do just one `ptr::write`

**Review by haraldh — APPROVED** [review]

**Pull request merged**